website/src/views/conceptMap.js

```diff
@@ -73,19 +75,47 @@ export default function ConceptMap() {
         let mounted = true;
         if (mounted) {
             setLoading(true);
-            // Fetch the student progressQueryString
-            apiv2.get(`/students/${selectedStudent}/progressquerystring`).then((res) => {
-                setStudentMastery(res.data);
+            // Fetch the student masterymapping
+            apiv2.get(`/students/${selectedStudent}/masterymapping`).then((res) => {
+                setMasteryMapping(res.data);
                 setLoading(false);
             });
         }
         return () => mounted = false;
     }, [selectedStudent])
 
+
+
+    async function fetchCMHTML() {
+        const url = `${window.location.origin}/progress`;
+        try {
+            const response = await fetch(url, {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(masteryMapping)
+            });
+            if (!response.ok) {
+                throw new Error(`HTTP error! Status: ${response.status}`);
+            }
+            const result = await response.text();
+            return result;
+        } catch (error) {
+            console.error("Error:", error);
+            return null;
+        }
+    }
+
+    fetchCMHTML().then(result => {
+        setCMhtml(result);
+    });
```

Comment:
This will be run for all renders.  I think you mean to use `useEffect` here.  I would recommend checking how we've done other API calls in the app for reference

Reply:
Changed! The reason I did not initially use `useEffect` for this is because `useEffect` cannot be `async`. However, I now realize that I can define an `async` function within `useEffect`.